dump/plugins/port.py

```diff
@@ -0,0 +1,77 @@
+from .executor import Executor
+from dump.match_infra import MatchEngine, MatchRequest
+from dump.helper import create_template_dict
+
+class Port(Executor):
+    """
+    Debug Dump Plugin for PORT Module
+    """
+    ARG_NAME = "port_name"
+    
+    def __init__(self):
+        self.match_engine = MatchEngine()
+        self.ret_temp = {}
+        self.ns = ''
+          
+    def get_all_args(self, ns=""):
+        req = MatchRequest(db="CONFIG_DB", table="PORT", key_pattern="*", ns=ns)
+        ret = self.match_engine.fetch(req)
+        all_ports = ret["keys"]
+        return [key.split("|")[-1] for key in all_ports]
+            
+    def execute(self, params):
+        self.ret_temp = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        port_name = params[Port.ARG_NAME]
+        self.ns = params["namespace"]
+        self.init_port_config_info(port_name)
+        self.init_port_appl_info(port_name)
+        port_asic_obj = self.init_asic_hostif_info(port_name)
+        self.init_asic_port_info(port_asic_obj)
+        self.init_state_port_info(port_name)
+        return self.ret_temp
+    
+    def add_to_ret_template(self, table, db, keys, err):
+        if not err and keys:
+            self.ret_temp[db]["keys"].extend(keys)
+            return True
+        else:
+            self.ret_temp[db]["tables_not_found"].extend([table])
+            return False
+    
+    def init_port_config_info(self, port_name):
```

Comment:
Can the below 3 init config/appl/state functions be merged into a single function?

Reply:
Any particular reason why? i believe this is more readable and also init_{module}_{db}_info is easy to identify. Will be certainly helpful to differentiate when dealing with modules which has complex use-cases.

Reply:
agreed, it is more readable like this.
my idea was if the init is done separately for each DB, then we could merge them.